content/docs/configuration-services.md

```diff
@@ -0,0 +1,327 @@
+
+In real scenarios, it is not recommend to completely replace a service. Instead, it is possible to extend a class and override only specific parts of it. Let's take a look at the `ArithmeticsModule` from the [arithmetics example](https://github.com/langium/langium/tree/main/examples/arithmetics).
+
+```Typescript
+export const ArithmeticsModule: Module<ArithmeticsServices, PartialLangiumServices & ArithmeticsAddedServices> = {
+    references: {
+        ScopeProvider: (services) => new ArithmeticsScopeProvider(services)
+    },
+    validation: {
+        ValidationRegistry: (services) => new ArithmeticsValidationRegistry(services),
+        ArithmeticsValidator: () => new ArithmeticsValidator()
+    }
+};
+```
+The `ArithmeticsValidator` is an [added service](#adding-new-services) while `ArithmeticsValidationRegistry` and `ArithmeticsScopeProvider` override `ValidationRegistry` and `DefaultScopeProvider`, respectively. When a call is made to `ValidationRegistry` or `ScopeProvider`, it will now create an instance of `ArithmeticsValidationRegistry` or `ArithmeticsScopeProvider` instead of the defaults `ValidationRegistry` or `DefaultScopeProvider`.
+
+In `ArithmeticsValidationRegistry`, we want to register our new `ArithmeticsValidator` to the `ValidationRegistry`.
+`ArithmeticsValidator` implements two checks, `checkDivByZero` and `checkNormalisable`:
+```Typescript
+    export class ArithmeticsValidator {
+    checkDivByZero(binExpr: BinaryExpression, accept: ValidationAcceptor): void {
+        ...
+    }
+
+    checkNormalisable(def: Definition, accept: ValidationAcceptor): void {
+        ...
+    }
+}
+``` 
+These two new checks need to be registered inside of the `ValidationRegistry`. Instead of directly modifying `ValidationRegistry` inside of the Langium code base, we extend `ValidationRegistry` with `ArithmeticsValidationRegistry` to implement our new functionalities:
+```Typescript
+export class ArithmeticsValidationRegistry extends ValidationRegistry {
+    constructor(services: ArithmeticsServices) {
+        super(services);
+        const validator = services.validation.ArithmeticsValidator;
+        const checks: LangiumGrammarChecks = {
+            BinaryExpression: validator.checkDivByZero,
+            Definition: validator.checkNormalisable
+        };
+        this.register(checks, validator);
+    }
+}
+```
+Inside of the `ArithmeticsValidationRegistry`, we call our `ArithmeticsValidator` via the `ServiceRegistry` with `const validator = services.validation.ArithmeticsValidator` which will create a new instance of `ArithmeticsValidator`. Then, we declare the `checks` to be registered and register them inside of the registry via the function `register` which is declared in the parent class. The `ArithmeticsValidationRegistry` only adds functionality to the `ValidationRegistry` but does not override any functionality from it.
```

Comment:
This part is quite specific to validation. Maybe extract that to a different section to keep "Overriding and Extending Services" more concise?

Reply:
I think dealing with the reorganization of the adding and overriding parts, as suggested by Christian will make this look more in place.